File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataSetModel.java
package com.mcreater.amclcore.api.mihoyo.genshin.model;

import com.google.gson.annotations.SerializedName;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class GenshinGachaDataSetModel {
    private int id;
    private String name;
    private String icon;
    private int level;
    @SerializedName("attr_id")
    private int attrId;
    @SerializedName("cat_id")
    private int catId;
    @SerializedName("wiki_url")
    private String wikiUrl;
    @SerializedName("default_tag")
    private List<GenshinGachaDataLevelModel> defaultTag;
    @SerializedName("extra_config")
    private GenshinGachaDataExtraConfigModel extraConfig;
}

File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataSetModel.kt
package com.mcreater.amclcore.api.mihoyo.genshin.model

import com.google.gson.annotations.SerializedName

data class GenshinGachaDataSetModel(
    var id: Int = 0,
    var name: String? = null,
    var icon: String? = null,
    var level: Int = 0,
    @SerializedName("attr_id") var attrId: Int = 0,
    @SerializedName("cat_id") var catId: Int = 0,
    @SerializedName("wiki_url") var wikiUrl: String? = null,
    @SerializedName("default_tag") var defaultTag: List<GenshinGachaDataLevelModel>? = null,
    @SerializedName("extra_config") var extraConfig: GenshinGachaDataExtraConfigModel? = null
)
